docs/content/blog/using-docker-in-bundles.md

```diff
@@ -77,138 +76,100 @@ required:
 ```
 
 We are going to focus on just using Docker for this blog post, but here is a [full
 working example for how to use Docker Compose in a
-bundle](https://github.com/deislabs/porter-docker-compose/tree/master/examples/compose).
+bundle](/src/examples/compose/).
 
 ### Use Docker
 
-Since we don't have a docker mixin, I am using the exec mixin to call the Docker CLI.
-For ease of testing, I put my commands into a helper script, `helpers.sh`. This lets
-me test out my commands locally without running the entire bundle.
-
-**helpers.sh**
-
-```bash
-#!/usr/bin/env bash
-set -euo pipefail
-
-whalesay() {
-  docker run --rm docker/whalesay:latest cowsay $1
-}
-
-install() {
-  whalesay "Hello World"
-}
-
-upgrade() {
-  whalesay "World 2.0"
-}
-
-uninstall() {
-  whalesay "Goodbye World"
-}
-
-# Call the requested function and pass the arguments as-is
-"$@"
-```
-
-Now in my porter.yaml I have very straightforward calls to the functions
-defined in my helpers.sh script for each action:
+In my porter.yaml, I can use the docker mixin to execute docker commands
+in my bundle:
 
 **porter.yaml**
 
 ```yaml
 name: whalesay
-version: 0.1.1
+version: 0.1.2
 description: "An example bundle that uses docker through the magic of whalespeak"
 registry: getporter
 
-dockerfile: Dockerfile.tmpl
-
 required:
-- docker
+  - docker
 
 parameters:
-- name: msg
-  description: a message for the whales to speak
-  type: string
-  default: "whale hello there!"
-  applyTo:
-  - say
+  - name: msg
+    description: a message for the whales to speak
+    type: string
+    default: "whale hello there!"
+    applyTo:
+      - say
 
 mixins:
-- exec
+  - docker
 
 install:
-- exec:
-    description: "Install Hello World"
-    command: ./helpers.sh
-    arguments:
-    - install
+  - docker:
+      run:
+        image: "docker/whalesay:latest"
+        rm: true
+        arguments:
+          - cowsay
+          - Hello World
 
 upgrade:
-- exec:
-    description: "World 2.0"
-    command: ./helpers.sh
-    arguments:
-    - upgrade
+  - docker:
+      run:
+        image: "docker/whalesay:latest"
+        rm: true
+        arguments:
+          - cowsay
+          - World 2.0
 
 say:
-- exec:
-    description: "Say Something"
-    command: ./helpers.sh
-    arguments:
-    - whalesay
-    - '"{{ bundle.parameters.msg }}"'
+  - docker:
+      run:
+        image: "docker/whalesay:latest"
+        rm: true
+        arguments:
+          - cowsay
+          - - "{{ bundle.parameters.msg }}"
 
 uninstall:
-- exec:
-    description: "Uninstall Hello World"
-    command: ./helpers.sh
-    arguments:
-    - uninstall
+  - docker:
+      run:
+        image: "docker/whalesay:latest"
+        rm: true
+        arguments:
+          - cowsay
+          - Goodbye World
 ```
 
-After I have tested the bundle, I used `porter publish` to push it up to `getporter/whalesay:v0.1.1`.
+After I have tested the bundle, I used `porter publish` to push it up to `getporter/whalesay:v0.1.2`.
```

Comment:
After I test the bundle and verify that it's ready for release, I use `porter publish` to push the new image `getporter/whalesay:v0.1.2` to the registry.

Reply:
suggesting this modification because pushing it up to the image name seems confusing.